pingcap/tidb
# br: backup: Add backup result into `backupMeta`

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/br/issues/1420

Problem Summary: Add the backup result into `backupMeta`

### What is changed and how it works?

What's Changed:
Add a member(backup_result) in brpb.proto. Write the `backupResult` when Flush the `backupmeta` into storage  


Tests 


Manual Test
```
./br backup full --pd "172.16.6.117:2379" --storage "local:///home/zak/backup_del" --check-requirements=false
```
The commd output in log
```
[2021/08/17 14:04:52.082 +08:00] [INFO] [collector.go:71] ["Full backup success summary"] [total-ranges=45] [ranges-succeed=45] [ranges-failed=0] [backup-checksum=4.453255632s] [backup-fast-checksum=7.247844ms] [backup-total-ranges=58] [backup-total-regions=77] [total-take=20.583101561s] [BackupTS=427079833093931010] [total-kv=15463846] [total-kv-size=1.859GB] [average-speed=90.29MB/s] ["backup data size(after compressed)"=496.8MB]
```
The tail part in `backupmeta` in storage
```
Release Version: v5.2.0-alpha-640-gd1b2452
Git Commit Hash: d1b2452c2e32a91168a87e8059fad4b7440d960b
Git Branch: add-backup-result
Go Version: go1.16.5
UTC Build Time: 2021-08-17 06:01:53
Race Enabled: false
<8a>^AÍ^B["Full backup"=success]["total-ranges"=45]["ranges-succeed"=45]["ranges-failed"=0]["backup-checksum"=4.453255632s]["backup-fast-checksum"=7.247844ms]["backup-total-ranges"=58]["backup-total-regions"=77]["total-take"=20.580926845s]["BackupTS"=427079833093931010]["total-kv"=15463846]["total-kv-size"=1.859GB]["average-speed"=90.3MB/s]
``` 

### Release note 

```release-note
Add the backup result into `backupMeta` in the storage
```
